### PR TITLE
Issue81 - Required mods check - Sanitize

### PR DESCRIPTION
**Pull request by ItsDyze**
Cleaned up a duplicate call to the config handler
Sanitized the mods input to exclude empty or whitespace strings

### CODE REVIEW

**Comment by ItsDyze** [comment]
Fixes #81

File: src/controllers/FikaClientController.ts [hunk 1]
```diff
@@ -25,8 +25,10 @@ export class FikaClientController {
     ) {
         const config = this.fikaConfig.getConfig();
 
-        this.requiredMods = new Set([...config.client.mods.required, "com.fika.core", "com.SPT.custom", "com.SPT.singleplayer", "com.SPT.core", "com.SPT.debugging"]);
-        this.allowedMods = new Set([...this.requiredMods, ...config.client.mods.optional, "com.bepis.bepinex.configurationmanager"]);
+        const sanitizeModList = (strArr: string[]): string[] => strArr.filter(str => str.trim() !== "");
```
Comment (ArchangelWTF) [inline src/controllers/FikaClientController.ts:28]: Could you make this a fully written method? I would rather not have in-line methods.

**Review by ArchangelWTF — CHANGES_REQUESTED** [review]
Could you target 3.10 with this PR? PR's for 3.9 aren't being accepted anymore.

File: src/controllers/FikaClientController.ts (hunk 1 above)
Comment (ItsDyze) [reply src/controllers/FikaClientController.ts]: Fixed, but perhaps the method would be better off in a Utils or something rather than the controller.

**Review by ArchangelWTF — APPROVED** [review]
Looks good! Thanks for the PR

**Comment by Lacyway** [comment]
Change doesn't work and now throws an error when the configuration is empty. Was this tested? Want to make sure I'm not doing anything wrong.

**Comment by ItsDyze** [comment]
Tested against empty values, no values and a required value. Perhaps I made a mistake in the refactoring of the method.
You mean without even the requiredMod property in the config?

**Comment by ItsDyze** [comment]
Looks fine with an empty array as default in the config.
Do you want me to do a separate PR to check for the absence of the key in the config? (I believe that would've crashed prior to my change as there was no check of the required nor mods keys)

`"mods": {
            "required": [],
            "optional": []
        },`

image

image

**Comment by ArchangelWTF** [comment]
I think the mistake was made in the check for the `allowedMods` size, it would obviously never be the size of 0 due to there being some default values set in there. Thus enforcing clients to always have those mods and only those mods loaded. (Not the intended behavior)

It's fixed in https://github.com/project-fika/Fika-Server/commit/cc8b3f05fd416d45573d85f80b19d17a1590498b

**Comment by Lacyway** [comment]
And thank you for the PR @ItsDyze!

**Pull request merged**